manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml

```diff
@@ -205,6 +205,10 @@ spec:
             - name: run-journal-vol
               mountPath: /run/log/journal
               readOnly: true
+          {{- if eq .Values.Azure.Cluster.Distribution "openshift" }}
+          securityContext:
+            privileged: true
```

Comment:
The plugin also need `privileged: true` if running on openshift. I see we're not setting a conditional for arc here: https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/129931f8ed26453cb22c4dd903e4c1147ff8cab0/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml#L88-L90

Is this being handled as part of the extension being enabled or we need to wrap this with a conditional `if eq .Values.Azure.Cluster.Distribution "openshift"` too?

Reply:
We don't need `privileged: true` for the plugin. I could fetch the secret just fine without it on ARO. We are only setting `runAsUser: 0` in the `securityContext`